### PR TITLE
Add post

### PR DESCRIPTION
**Pull request by wobblesphere**

### CODE REVIEW

File: src/Home/Home.js [hunk 1]
```diff
@@ -38,7 +41,7 @@ class Home extends Component {
 
 function mapStateToProps(state){
     return{
-        isPageMaskShown: state.AppReducer.get('isPageMaskShown'),
+        isPageMaskShown: state.ModalReducer.get('isPageMaskShown'),
```
Comment (alixander) [inline src/Home/Home.js]: Reminder -- you don't need to put `Reducer` in the name, just `state.modal` would be fine

File: src/Modals/ModalNewForm/ModalNewForm.css [hunk 2]
```diff
@@ -1,20 +1,14 @@
-@import url('https://fonts.googleapis.com/css?family=Poiret+One');
+/* @import url('https://fonts.googleapis.com/css?family=Poiret+One'); */
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.css]: delete

File: src/Modals/ModalNewForm/ModalNewForm.css [hunk 3]
```diff
@@ -71,3 +66,29 @@ input{
     font-weight: bolder;
 }
 
+.button_submit, .button_cancel{
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.css]: sometimes it's better to just have multiple classes applied to divs. For example, in this instance, it seems you want a generic `.button` class, and then `.button_cancel` and `.button_submit` can be more specific, and the div for button_cancel would have both `.button .button_cancel` classes

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import "./ModalNewForm.css";
+import { connect } from 'react-redux';
+import Actions from "../../Actions/actions";
+
+class PostForm extends Component {
+    constructor() {
+        super();
+        this.state = {
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js:9]: ?

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        }
+    }
+
+    getDate(){
+        let date = new Date();
+        var month = date.getUTCMonth() + 1;
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js]: var?

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        var day = date.getUTCDate();
+        var year = date.getUTCFullYear();
+        return month + "/" + day + "/" +  year;
+    }
+
+    onChangeInput(type, answer){
+        this.setState({
+            [type]: answer,
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js:33]: nice job figuring this one out!

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+           'name': this.state.name,
+           'address': address,
+           'note': this.state.note
+       }
+       this.props.toggleModal('newForm', false)
+       return postInfo
+    }
+
+    render() {
+        if(this.props.modalShown.includes('newForm')){
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js]: See how this is nested? It makes it look uglier. Instead, it's better to do

```
if (!condition) {
  return null
}
...rest of code
```

File: src/Modals/ModalSignUp/ModalSignUp.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import "./ModalSignUp.css";
+import { connect } from 'react-redux';
+import OutsideClick from '../../OutsideClick';
+// import Actions from "../../Actions/actions";
+
+class ModalSignUp extends Component {
+    render(){
+        if(this.props.modalShown.includes('signUp')){
```
Comment (alixander) [inline src/Modals/ModalSignUp/ModalSignUp.js]: same here. don't nest all this

File: src/Modals/ModalSignUp/ModalSignUp.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                </OutsideClick>
+                )
+        }
+        return null
+    }
+}
+
+function mapStateToProps(state) {
+    return{
+        modalShown: state.ModalReducer.getIn(['modalShown']),
```
Comment (alixander) [inline src/Modals/ModalSignUp/ModalSignUp.js]: don't need getIn

File: src/NavBar/NavBar.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react';
 import "./NavBar.css";
 import StyleMenu from './StyleMenu/StyleMenu.js';
 import FilterMenu from './FilterMenu/FilterMenu.js';
-import { NavLink } from 'react-router-dom';
+// import { NavLink } from 'react-router-dom';
```
Comment (alixander) [inline src/NavBar/NavBar.js:5]: remove if u don't need

File: src/reducers/ModalReducer.js [hunk 11]
```diff
@@ -2,6 +2,7 @@ import { Map, List } from 'immutable';
 
 const INITIAL_STATE = Map({
     'modalShown': List(['zipcode']),
+    'isPageMaskShown': true,
```
Comment (alixander) [inline src/reducers/ModalReducer.js:5]: is this ever updated?

File: src/reducers/postBoardReducer.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
-import { Map } from 'immutable';
+import { Map, List } from 'immutable';
 
 const INITIAL_STATE = Map({
-    posts: [
+    posts: List([
```
Comment (alixander) [inline src/reducers/postBoardReducer.js]: nice

File: src/reducers/postBoardReducer.js [hunk 13]
```diff
@@ -42,6 +42,8 @@ function PostBoardReducer (state = INITIAL_STATE, action) {
             return state.set('displayStyle', action.data)
         case('UPDATE_CLICKED_POST'):
             return state.set('clickedListPost', action.data)
+        case('ADD_NEW_POST'):
+            return state.updateIn(['posts'], list => list.push(action.data))
```
Comment (alixander) [inline src/reducers/postBoardReducer.js]: updateIn?

**Review by alixander — CHANGES_REQUESTED** [review]
Very good! I notice you figuring out stuff on your own =)

File: src/Modals/ModalNewForm/ModalNewForm.js (hunk 4 above)
Comment (wobblesphere) [reply src/Modals/ModalNewForm/ModalNewForm.js:9]: this state is actually being set everytime an input has changed; so its not empty! LOL

File: src/NavBar/NavBar.js (hunk 10 above)
Comment (wobblesphere) [reply src/NavBar/NavBar.js:5]: I will need it though

File: src/reducers/ModalReducer.js (hunk 11 above)
Comment (wobblesphere) [reply src/reducers/ModalReducer.js:5]: yes, everytime a modal is being toggle, it updates mask at the same time

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+
+    handleSubmit(){
+        const address = `${this.state.address_street} ${this.state.address_apt}, ${this.state.address_city}`
+        const postInfo = {
+           'date': this.getDate(),
+           'name': this.state.name,
+           'address': address,
+           'note': this.state.note
+       }
+       if(this.isSubmitButtonEnable()){
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js]: Do you need to check this here? The user wouldn't be able to get to this code unless submit button was enabled right?

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+        this.props.addNewPost(postInfo)
+        this.props.toggleModal('newForm', false)
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js]: This is something that could be done through sagas! You're doing 2 actions when submit is clicked right? But toggling the modal isn't the **main** action, it's a side effect. So it'd be cleaner to have a saga watch for whenever a new post is added, close the modal.

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+        this.clearState();
+       }
+       return null;
+    }
+
+    checkInput(value){
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js]: This is returning a string but the name implies an action. A better name would be `getErrorMessage`

File: src/Modals/ModalNewForm/ModalNewForm.js [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+        return "value_invalid"
+    }
+
+    render() {
+        if(!this.props.modalShown.includes('newForm')){ 
+            return null
+        }
+        return(
+        <form>
+            <div className="newPostForm">
```
Comment (alixander) [inline src/Modals/ModalNewForm/ModalNewForm.js]: This is too long, I'd recommend splitting it into multiple renders.

e.g.

```
{this.renderHeader()}
{this.renderQuestion()}
...
```

File: src/Modals/ModalSignUp/ModalSignUp.js [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+    render(){
+        if(!this.props.modalShown.includes('signUp')){ 
+            return null
+        }
+        return(
+            <OutsideClick>
+                <div className="form_signUp">
+                    <div className="form_signUp_header">Create an account: </div>
+                    <div className="signUp_username">
+                        <div className="create_username">Username: </div>
```
Comment (alixander) [inline src/Modals/ModalSignUp/ModalSignUp.js:17]: username_label is probably a better name

File: src/Modals/ModalSignUp/ModalSignUp.css [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+
+.signUp_username, .signUp_email{
+    width: 100%;
+    height: 40px;
+}
+
+.signUp_username{
+    margin-bottom: 10px;
+}
+.input_username, .input_email{
```
Comment (alixander) [inline src/Modals/ModalSignUp/ModalSignUp.css:34]: you have a lot of rules that apply to both. Remember to use the trick where you can assign multiple class names to a single div. 

So a `signup_modal_label` class will have all these rules, then you can have more specific `signup_modal_label--username` for rules specific to username

File: src/NavBar/NavBar.js (hunk 10 above)
Comment (alixander) [reply src/NavBar/NavBar.js:5]: -__-

File: src/Post/Post-List/Post-List.css [hunk 20]
```diff
@@ -50,18 +46,16 @@
...
-    left: 30%;
-    background:red;
-}
 
 .icon_circle {
     color: green;
     font-size: 10px;
+}
+
+.postList_name:hover, .postList_date:hover, .postList_address:hover, .postList_note:hover {
```
Comment (alixander) [inline src/Post/Post-List/Post-List.css]: Another instance where it's cleaner to use multiple class names. applying a `postList_label` to each of those divs

File: src/PostBoard/PostBoard.js [hunk 21]
```diff
@@ -49,8 +49,8 @@ class PostBoard extends Component {
 
 function mapStateToProps(state){
     return{
-        posts: state.PostBoardReducer.get('posts'),
-        displayStyle: state.PostBoardReducer.get('displayStyle')
+        postsInfo: state.PostBoard.get('posts'),
```
Comment (alixander) [inline src/PostBoard/PostBoard.js]: i'm not sure I agree with the rename

File: src/PostBoard/PostBoard.js [hunk 22]
```diff
@@ -33,7 +33,7 @@ class PostBoard extends Component {
 
     producePosts(){
         if(this.props.displayStyle === 'Gallery'){
-            return this.producePostsGalery()
+            return this.producePostsGallery()
```
Comment (alixander) [inline src/PostBoard/PostBoard.js]: conventional to call things that give back HTML elements `renderX()`

File: src/reducers/postBoardReducer.js [hunk 23]
```diff
@@ -42,6 +41,8 @@ function PostBoardReducer (state = INITIAL_STATE, action) {
             return state.set('displayStyle', action.data)
         case('UPDATE_CLICKED_POST'):
             return state.set('clickedListPost', action.data)
+        case('ADD_NEW_POST'):
+            return state.update('posts', list => list.push(action.data));
```
Comment (alixander) [inline src/reducers/postBoardReducer.js]: ;)

**Review by alixander — CHANGES_REQUESTED** [review]
Very good!

**Pull request merged**